### PR TITLE
Blogpost

### PR DESCRIPTION
**Pull request by qingyun-wu**
## Why are these changes needed?

Adding a blog post on FLAML's 1 million Downloads and a glimpse into FLAML v2

## Checks

- I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/blog/2023-05-07-1M-milestone/index.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+FLAML is designed for easy extensibility, allowing users to add custom learners or metrics. The level of customization ranges from minimal (providing only training data and task type as input) to full (tuning a user-defined function). This flexibility and support for easy customization have led to FLAML's adoption in various domains, including security, finance, marketing, engineering, supply chain, insurance, and healthcare, delivering highly accurate results.
+
+## Embracing Large Language Models in FLAML v2
+As large language models continue to reshape the AI ecosystem, FLAML is poised to adapt and grow alongside these advancements. Recognizing the importance of large language models, we have recently incorporated an autogen package into FLAML, and are committed to focusing our collective efforts on addressing the unique challenges that arise in LMOps (Large Language Model Operations).
+
+In its current iteration, FLAML offers support for model selection and inference parameter tuning for large language models. We are actively working on the development of new features, such as agent-based LLM operations and enhanced communication, to further expand FLAML's capabilities.
```
[the remark this answers is not in the document]
Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx:31]: Thanks. Revise.

**Comment by qingyun-wu** [comment]
TODO: To add specific industry partners that used FLAML. 
Could you provide the information you know about this? @sonichi @thinkall

File: website/blog/2023-05-07-1M-milestone/index.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## FLAML's Journey: Past Achievements and Milestones
+
+### Bring AutoML to Ones Fingertips
+FLAML offers an off-the-shelf AutoML solution that enables users to quickly discover high-quality models or configurations for common ML/AI tasks. By automatically selecting models and hyperparameters for training or inference, FLAML saves users time and effort. FLAML has significantly reduced development time for developers and data scientists alike, while also providing a convenient way to integrate new algorithms into the pipeline, enabling easy extensions and large-scale parallel tuning. These features make FLAML a valuable tool in R&D efforts for many enterprise users.
+
+### Speed and Efficiency: The FLAML Advantage
+What sets FLAML apart from other AutoML libraries is its efficiency. Both the Tune and AutoML modules are fast, thanks to efficient hyperparameter optimization and model selection methods. FLAML is also capable of handling large search spaces with heterogeneous evaluation costs, complex constraints, guidance, and early stopping.
```
[the remark this answers is not in the document]
Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx:23]: Revised.

File: website/blog/2023-05-07-1M-milestone/index.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+tags: [LLM, LLMOps, FLAMLv2]
+---
+
+**TL;DR:**
+* **Celebrating FLAML's milestone: 1 million downloads**
+* **Introducing Large Language Model (LLM) support in the upcoming FLAML v2**
+
+
+This week, FLAML has reached a significant milestone: 1 million downloads. Originating as an intern research project within Microsoft Research, FLAML has grown into an open-source library used widely across the industry and supported by an active community.
+As we celebrate this milestone, we want to recognize the passionate contributors and users who have played an essential role in molding FLAML into the flourishing project it is today. Our heartfelt gratitude goes out to each of you for your unwavering support, constructive feedback, and innovative contributions that have driven FLAML to new heights.
```
Comment (sonichi) [inline website/blog/2023-05-07-1M-milestone/index.mdx:13]: A big shoutout to our industrial collaborators from Azure Compute, Azure Data (@thinkall), Azure Machine Learning, Microsoft 365, ML.NET, Vowpal Wabbit, Anyscale, Databricks, and Wise; and academic collaborators from MIT, Penn State University, Stevens Institute of Technology, Tel Aviv University, Texas A & M University, University of Manchester, University of Washington, and The Chinese University of Hong Kong etc.

File: website/blog/2023-05-07-1M-milestone/index.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+### Bring AutoML to One's Fingertips
+FLAML offers an off-the-shelf AutoML solution that enables users to quickly discover high-quality models or configurations for common ML/AI tasks. By automatically selecting models and hyperparameters for training or inference, FLAML saves users time and effort. FLAML has significantly reduced development time for developers and data scientists alike, while also providing a convenient way to integrate new algorithms into the pipeline, enabling easy extensions and large-scale parallel tuning. These features make FLAML a valuable tool in R&D efforts for many enterprise users.
+
+### Speed and Efficiency: The FLAML Advantage
+What sets FLAML apart from other AutoML libraries is its efficiency. Both [`flaml.tune`](https://microsoft.github.io/FLAML/docs/Use-Cases/Tune-User-Defined-Function) and [flaml.AutoML](https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML) modules are fast, thanks to efficient hyperparameter optimization and model selection methods. FLAML is also capable of handling large search spaces with heterogeneous evaluation costs, complex constraints, guidance, and early stopping.
```
Comment (sonichi) [inline website/blog/2023-05-07-1M-milestone/index.mdx:23]: In Python, a module refers to a .py file. So I feel odd when seeing "modules".

File: website/blog/2023-05-07-1M-milestone/index.mdx (hunk 3 above)
Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx:13]: Added. thanks!

File: website/blog/2023-05-07-1M-milestone/index.mdx (hunk 4 above)
Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx]: Removed the word "modules". Thanks!

File: website/blog/2023-05-07-1M-milestone/index.mdx (hunk 3 above)
Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx:13]: okay, @thinkall, could you double check?

Comment (thinkall) [reply website/blog/2023-05-07-1M-milestone/index.mdx:13]: > okay, @thinkall, could you double check?

Oh, yes, Azure Data is correct. Moreover, Azure Synapse is even more precise and is at the same organizational level as Azure Machine Learning.

Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx:13]: Thanks! Updated!

**Comment by EgorKraevTransferwise** [comment]
It would be really cool to have a paragraph on all the different task types FLAML can handle, with links to examples in the docs. As is, I feel the variety of what FLAML can already do doesn't quite come across.

Also, in the FLAML advantage paragraph you could add "It even supports hierarchical search spaces, for example first choosing an estimator and then sampling from the hyperparameter space specific to that estimator."

**Comment by qingyun-wu** [comment]
> It would be really cool to have a paragraph on all the different task types FLAML can handle, with links to examples in the docs. As is, I feel the variety of what FLAML can already do doesn't quite come across.
> 
> Also, in the FLAML advantage paragraph you could add "It even supports hierarchical search spaces, for example first choosing an estimator and then sampling from the hyperparameter space specific to that estimator."

Thank you for your valuable input Egor! I will add a paragraph on the task that FLAML could handle and the suggested content on hierarchical search space.

File: website/blog/2023-05-07-1M-milestone/index.mdx (hunk 3 above)
Comment (thinkall) [reply website/blog/2023-05-07-1M-milestone/index.mdx:13]: As our PM commented today, "Azure Synapse Analytics" is better.

Comment (qingyun-wu) [reply website/blog/2023-05-07-1M-milestone/index.mdx:13]: Thanks @thinkall ! Updated.

**Comment by qingyun-wu** [comment]
> > It would be really cool to have a paragraph on all the different task types FLAML can handle, with links to examples in the docs. As is, I feel the variety of what FLAML can already do doesn't quite come across.
> > Also, in the FLAML advantage paragraph you could add "It even supports hierarchical search spaces, for example first choosing an estimator and then sampling from the hyperparameter space specific to that estimator."

Thank you for your valuable input Egor! I have added content on the tasks that FLAML could handle and some content on hierarchical search space. I am not elaborating in detail in order to keep the blog post concise and easy to digest.

**Review by sonichi — APPROVED** [review]

**Review by thinkall — APPROVED** [review]

**Pull request merged**